main/zoe/guide/contracts/simple-exchange.md

```diff
@@ -2,19 +2,208 @@
 
 <Zoe-Version/>
 
-##### [View the code on Github](https://github.com/Agoric/agoric-sdk/blob/958a2c0a3dec38bdba2234934119ea2c28958262/packages/zoe/src/contracts/simpleExchange.js) (Last updated: 4/22/2020)
-##### [View all contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)
+##### [View all example contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)
 
 The "simple exchange" is a very basic, un-optimized exchange. The
 simple exchange has an order book for one asset, priced in a second
 asset. The order book is naively kept as an array that is iterated over
 to look for matches every time a new order arrives.
+An unlimited number of parties can create new orders or accept existing
+orders. The notifier allows callers to find the current list of orders.
 
-## Offer rules
+SimpleExchange uses `Asset` and `Price` as its keywords. The contract treats
+them symmetrically. New offers can be created and existing offers can be 
+accepted in either direction.
+```js
+{ give: { 'Asset', simoleans(5) }, want: { 'Price', quatloos(3) } }
+{ give: { 'Price', quatloos(8) }, want: { 'Asset', simoleans(3) } }
+```
+
+The `Asset` is an exact amount to be exchanged, while the
+`Price` is a limit that may be improved on. SimpleExchange does
+not partially fill orders.
+
+The invitation returned on contract installation is the same as what
+is returned by calling `publicAPI.makeInvite()`.
+
+## Setting things up
+
```

Comment:
The above is a good description of the external interface. I would make the description of the internals much shorter. The high points are (1) it keeps two lists of offerHandles for buys and sells, (2) each new offer is compared to all the existing offers looking for a match with `satisfies()` and `satisfiedBy()`, (3) if a match is found `swap(offerHandle, iHandle)` is called, (4) otherwise the new offer is added to the appropriate list, and (5) the contract uses `zcf.initPublicAPI()` to provide three functions that anyone can call to get the status of an offer, to get a notifier to see the book, or to make a new invitation.